packages/api/utils/validateRequestShape.ts

```diff
@@ -0,0 +1,35 @@
+import { APIGatewayProxyEventPathParameters } from "aws-lambda";
+import createError from "http-errors";
+import { ApiGatewayEventParsed } from "./eventHandlingTool";
+import { SetupSuccess } from "./response";
+import { isValidUuidV4 } from "./validation";
+
+/**
+ * Check if incoming POST Request passes basic shape validation
+ *
+ * This shape validation checks the pathParameter to ensure it is not null or undefined, and that is
+ * a valid UUIDv4.
+ *
+ * @param event - The incoming API Gateway Request proxied to Lambda
+ * @param extraValidators - Additional validators that check whether the body is a valid object of Type T
+ * @returns SetUpSuccess object if event passes validation, otherwise it throws an error
+ */
+export function validateRequestShape<T>(
+  event: ApiGatewayEventParsed<T>,
+  ...extraValidators: Array<(obj: unknown) => obj is T>
+): SetupSuccess<T> {
+  if (!Object.values(event.pathParameters as APIGatewayProxyEventPathParameters).every(isValidUuidV4)) {
+    throw createError(404, "Shape validation failed, invalid UUIDv4");
```

Comment:
Let's make this more generic, like _invalid path parameter_ or _invalid identifier_.  Even in the new API spec the only place we specify uuid is the S3 object key.

Reply:
Sure, it can be changed. Originally the message was linked to the `switch` statement in `errorHandlingMiddleware`, but is currently being caught with `error.name && error.name.startsWith("EntityNotFoundError")` instead of the `switch` statement in PR #500.

Reply:
If this isn't making its way up to the user, nevermind.  What you've got in #500 in `errorHandlingMiddleware` 
 returning `NO_SUCH_PORTFOLIO_OR_APPLICATION` is fine.

Reply:
Made the change, shouldn't make a difference.